src/mad_spark_alt/core/prompt_classifier.py

```
@@ -0,0 +1,435 @@
+            confidence *= 1.1  # Boost for matching multiple types (indicates richness)
+        
+        # Ensure confidence is between 0 and 1
+        confidence = min(max(confidence, 0.0), 1.0)
+        
+        # Lower minimum confidence threshold to 15%
+        if confidence < 0.15:
+            return QuestionType.UNKNOWN, confidence
+        
+        return best_type, confidence
+    
+    def _detect_complexity(self, question_lower: str, question_length: int) -> ComplexityLevel:
```

Ruff (0.12.2)

`_detect_complexity` is too complex (13 > 10)

(C901)